core/Pos/Binary/Class/Core.hs

```diff
@@ -1,171 +1,586 @@
-{-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE PolyKinds           #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies        #-}
+{-# LANGUAGE TypeOperators       #-}
+module Pos.Binary.Class.Core
+    ( Bi(..)
+    , encodeBinary
+    , decodeBinary
+    , enforceSize
+    , matchSize
+    -- * CBOR re-exports
+    , encodeListLen
+    , decodeListLen
+    , decodeListLenOf
+    , Encoding
+    , Decoder
+    -- * GHC-Generics-based encoding & decoding
+    , genericEncode
+    , genericDecode
+    ) where
 
-#include "MachDeps.h"
+import           Codec.CBOR.Decoding
+import           Codec.CBOR.Encoding
+import           GHC.Generics
+import qualified Data.Binary                 as Binary
+import qualified Data.ByteString             as BS
+import qualified Data.ByteString.Lazy        as BS.Lazy
+import           Data.Fixed                  (Fixed(..), Nano)
+import qualified Data.HashMap.Strict         as HM
+import qualified Data.HashSet                as HS
+import qualified Data.Map                    as M
+import qualified Data.Set                    as S
+import           Data.Tagged
+import qualified Data.Text                   as Text
+import           Data.Time.Units             (Microsecond, Millisecond)
+import qualified Data.Vector                 as Vector
+import qualified Data.Vector.Generic         as Vector.Generic
+import           Serokell.Data.Memory.Units  (Byte, fromBytes, toBytes)
+import           Universum                   hiding (foldr)
+import qualified Universum
 
--- | Bi typeclass and most basic functions.
```

Comment:
Why did you remove it? We tend to have a Haddock comment on top of every modules (though sometimes people don't add them).